src/symmetry/rotation.hpp

```diff
@@ -197,40 +197,39 @@ inline vector3d<double> euler_angles(matrix3d<double> const& rot__)
     auto rit = inverse(transpose(rot__));
     for (int i = 0; i < 3; i++) {
         for (int j = 0; j < 3; j++) {
-            rit(i, j) = (rit(i, j) + rot__(i, j)) / 2.0;
-            //if (std::abs(rot__(i, j) - rit(i, j)) > 1e-8) {
-            //    std::stringstream s;
-            //    s << "rotation matrix is not unitary" << std::endl
-            //      << "initial symmetry matrix:" << std::endl
-            //      << rot__ << std::endl
-            //      << "inverse transpose matrix:" << std::endl
-            //      << rit;
-            //    RTE_THROW(s);
-            //}
+            if (std::abs(rot__(i, j) - rit(i, j)) > tolerance__) {
```

Comment:
This was a bad idea to average the input potentially non-unitary matrix. Not all unitary transformations are rotations and the problem of having a wrong symmetry here is not solved. euler_angles() should check for strict unitary matrix